packages/gestalt-codemods/_next/select-list-options-to-children.js

```diff
@@ -0,0 +1,165 @@
+  if (!localIdentifierName) {
+    return null;
+  }
+
+  const transform = src
+    .find(j.JSXElement)
+    .forEach((jsxElement) => {
+      const { node } = jsxElement;
+
+      // This isn't SelectList, bail
+      if (!localIdentifierName.includes(node.openingElement.name.name)) {
+        return null;
+      }
+
+      const attrs = node.openingElement.attributes;
+
+      // Spread props can't be codemodded, prompt for manual update
+      if (attrs.some((attr) => attr.type === 'JSXSpreadAttribute')) {
+        throw new Error(
+          `Remove dynamic SelectList properties and rerun codemod. Location: ${file.path} @line: ${node.loc.start.line}`,
+        );
+      }
+
+      // If the options are defined inline (array literal)
+      let options;
+      // If the options are a variable defined elsewhere
+      let optionsVariableName;
+
+      // Map through existing attributes, grabbing the value for `options` and removing it as a prop
+      const newAttrs = attrs
+        .map((attr) => {
+          const propName = attr?.name?.name;
+
+          // Not `options`, bail
+          if (propName !== 'options') {
+            return attr;
+          }
+
+          // options defined inline
+          if (
+            attr?.value?.type === 'JSXExpressionContainer' &&
+            attr?.value?.expression?.type === 'ArrayExpression'
+          ) {
+            options = attr?.value?.expression?.elements;
+            return null;
+          }
+
+          // options defined with a variable
+          if (
+            attr?.value?.type === 'JSXExpressionContainer' &&
+            attr?.value?.expression?.type === 'Identifier'
+          ) {
+            optionsVariableName = attr?.value?.expression?.name;
+            return null;
+          }
+
+          // Something weird is going on here, this should never be reached
+          // eslint-disable-next-line no-console
+          console.log(
+            `${node.openingElement.name.name} components with "option" prop must be converted to use ${node.openingElement.name.name}.Option manually. Location: ${file.path} @line: ${node.loc.start.line}`,
+          );
+          return null;
+        })
+        .filter(Boolean);
+
+      if (options || optionsVariableName) {
+        fileHasModifications = true;
+
+        const arrowFuncArgName = 'item';
+        const jsxElementName = localIdentifierName[0] ?? 'SelectList';
+
+        // Reassign the new attributes, which no longer contain `options`
+        node.openingElement.attributes = newAttrs;
+        // We're adding children, so this is no longer self-closing and we need a closing element
+        node.openingElement.selfClosing = false;
+        node.closingElement = j.jsxClosingElement(j.jsxIdentifier(jsxElementName));
+
+        const mapTarget = options ? j.arrayExpression(options) : j.identifier(optionsVariableName);
+
+        // Add the existing options array as mapped children
+        node.children = [
```

Comment:
hoo boy 😅

Reply:
wow, thanks for writing this. Good reference for the future.  I would have just thrown an error on detection ha

Reply:
I finally learned to read the "documentation" (haha) to figure this stuff out. Let me know the next time you're writing something like this and I'm happy to pair with you to teach you. It's pretty simple once you figure it out, just kinda tedious.